From 00eef69fc190c53cf544dbee2e3fa305cde9395a Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:47 -0400
Subject: [virt] kvm/nvmx: Advertise IA32_PAT in VM exit control

Message-id: <1406040016-3289-73-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86151
O-Subject: [RHEL7 PATCH v2 072/221] KVM: nVMX: Advertise IA32_PAT in VM exit control
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Arthur Chunqi Li <yzt356@gmail.com>

Advertise VM_EXIT_SAVE_IA32_PAT and VM_EXIT_LOAD_IA32_PAT.

Signed-off-by: Arthur Chunqi Li <yzt356@gmail.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit c0dfee582ef22c35dd4c208e502aa72cab646594)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index f2508d9..8108127 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2196,13 +2196,15 @@ static __init void nested_vmx_setup_ctls_msrs(void)
   * If bit 55 of VMX_BASIC is off, bits 0-8 and 10, 11, 13, 14, 16 and
   * 17 must be 1.
   */
+ rdmsr(MSR_IA32_VMX_EXIT_CTLS,
+  nested_vmx_exit_ctls_low, nested_vmx_exit_ctls_high);
  nested_vmx_exit_ctls_low = VM_EXIT_ALWAYSON_WITHOUT_TRUE_MSR;
  /* Note that guest use of VM_EXIT_ACK_INTR_ON_EXIT is not supported. */
+ nested_vmx_exit_ctls_high &=
 #ifdef CONFIG_X86_64
- nested_vmx_exit_ctls_high = VM_EXIT_HOST_ADDR_SPACE_SIZE;
-#else
- nested_vmx_exit_ctls_high = 0;
+  VM_EXIT_HOST_ADDR_SPACE_SIZE |
 #endif
+  VM_EXIT_LOAD_IA32_PAT | VM_EXIT_SAVE_IA32_PAT;
  nested_vmx_exit_ctls_high |= (VM_EXIT_ALWAYSON_WITHOUT_TRUE_MSR |
           VM_EXIT_LOAD_IA32_EFER);
 
-- 
1.7.1